lib/app/features/feed/views/pages/article_details_page/article_details_page.dart

```diff
@@ -27,9 +45,36 @@ class ArticleDetailsPage extends ConsumerWidget {
       return const SizedBox.shrink();
     }
 
+    final scrollController = useScrollController();
+
+    useEffect(
+      () {
+        void onScroll() {
+          if (!scrollController.hasClients) return;
+
+          final maxScroll = scrollController.position.maxScrollExtent;
+          final currentScroll = scrollController.offset;
+          final viewportWidth = MediaQuery.of(context).size.width;
```

Comment:
`MediaQuery.sizeOf(context).width`
could you use like this to to reduce render count?